❌ **Build Blocked:** Step 1/1 (`Address review feedback`) was **not committed** because build errors could not be resolved after 5 fix attempts + full code regeneration.

This step needs manual review or will be addressed in a follow-up.

<details>
<summary>Build Errors (last attempt)</summary>

```
C:\Agents\principalengineer-16dbc662e38e45b18a5da8a501c41114\ReportingDashboard\src\AgentSquad.Runner\Models\Project.cs(3,14): error CS0101: The namespace 'AgentSquad.Runner.Models' already contains a definition for 'Project' [C:\Agents\principalengineer-16dbc662e38e45b18a5da8a501c41114\ReportingDashboard\src\AgentSquad.Runner\AgentSquad.Runner.csproj]
C:\Agents\principalengineer-16dbc662e38e45b18a5da8a501c41114\ReportingDashboard\src\AgentSquad.Runner\Models\Milestone.cs(3,14): error CS0101: The namespace 'AgentSquad.Runner.Models' already contains a definition for 'Milestone' [C:\Agents\principalengineer-16dbc662e38e45b18a5da8a501c41114\ReportingDashboard\src\AgentSquad.Runner\AgentSquad.Runner.csproj]
C:\Agents\principalengineer-16dbc662e38e45b18a5da8a501c41114\ReportingDashboard\src\AgentSquad.Runner\Models\WorkItem.cs(3,14): error CS0101: The namespace 'AgentSquad.Runner.Models' already contains a definition for 'WorkItem' [C:\Agents\principalengineer-16dbc662e38e45b18a5da8a501c41114\ReportingDashboard\src\AgentSquad.Runner\AgentSquad.Runner.csproj]
C:\Agents\principalengineer-16dbc662e38e45b18a5da8a501c41114\ReportingDashboard\src\AgentSquad.Runner\Models\Project.cs(3,14): error CS0101: The namespace 'AgentSquad.Runner.Models' already contains a definition for 'Project' [C:\Agents\principalengineer-16dbc662e38e45b18a5da8a501c41114\ReportingDashboard\src\AgentSquad.Runner\AgentSquad.Runner.csproj]
C:\Agents\principalengineer-16dbc662e38e45b18a5da8a501c41114\ReportingDashboard\src\AgentSquad.Runner\Models\Milestone.cs(3,14): error CS0101: The namespace 'AgentSquad.Runner.Models' already contains a definition for 'Milestone' [C:\Agents\principalengineer-16dbc662e38e45b18a5da8a501c41114\ReportingDashboard\src\AgentSquad.Runner\AgentSquad.Runner.csproj]
C:\Agents\principalengineer-16dbc662e38e45b18a5da8a501c41114\ReportingDashboard\src\AgentSquad.Runner\Models\WorkItem.cs(3,14): error CS0101: The namespace 'AgentSquad.Runner.Models' already contains a definition for 'WorkItem' [C:\Agents\principalengineer-16dbc662e38e45b18a5da8a501c41114\ReportingDashboard\src\AgentSquad.Runner\AgentSquad.Runner.csproj]
```
</details>

--- a/AgentSquad.sln
+++ b/AgentSquad.sln
@@ -1,0 +1,24 @@
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio Version 17
+VisualStudioVersion = 17.8.34309.116
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7B33}") = "AgentSquad.Runner", "src\AgentSquad.Runner\AgentSquad.Runner.csproj", "{6B4D8C2E-1F9A-4D7C-9E3B-2A1F5C8D9E0A}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7B33}") = "AgentSquad.Runner.Tests", "src\AgentSquad.Runner.Tests\AgentSquad.Runner.Tests.csproj", "{7C5D9D3F-2G0B-5E8D-0F4C-3B2G6D9E0F1B}"
+EndProject
+Global
+	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Debug|Any CPU = Debug|Any CPU
+		Release|Any CPU = Release|Any CPU
+	EndGlobalSection
+	GlobalSection(ProjectConfigurationPlatforms) = postSolution
+		{6B4D8C2E-1F9A-4D7C-9E3B-2A1F5C8D9E0A}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{6B4D8C2E-1F9A-4D7C-9E3B-2A1F5C8D9E0A}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{6B4D8C2E-1F9A-4D7C-9E3B-2A1F5C8D9E0A}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{6B4D8C2E-1F9A-4D7C-9E3B-2A1F5C8D9E0A}.Release|Any CPU.Build.0 = Release|Any CPU
+		{7C5D9D3F-2G0B-5E8D-0F4C-3B2G6D9E0F1B}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{7C5D9D3F-2G0B-5E8D-0F4C-3B2G6D9E0F1B}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{7C5D9D3F-2G0B-5E8D-0F4C-3B2G6D9E0F1B}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{7C5D9D3F-2G0B-5E8D-0F4C-3B2G6D9E0F1B}.Release|Any CPU.Build.0 = Release|Any CPU
+	EndGlobalSection
+EndGlobal

--- a/src/AgentSquad.Runner/AgentSquad.Runner.csproj
+++ b/src/AgentSquad.Runner/AgentSquad.Runner.csproj
@@ -3,11 +3,18 @@
   <PropertyGroup>
     <TargetFramework>net8.0</TargetFramework>
     <Nullable>enable</Nullable>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Version>1.0.0</Version>
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Components.Web" Version="8.0.6" />
+    <PackageReference Include="System.Net.Http.Json" Version="8.0.0" />
     <PackageReference Include="MudBlazor" Version="7.0.0" />
-    <PackageReference Include="ChartJS.Blazor" Version="4.1.2" />
+    <PackageReference Include="Microsoft.Extensions.Configuration" Version="8.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="8.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="8.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Options.ConfigurationExtensions" Version="8.0.0" />
   </ItemGroup>
 
   <ItemGroup>

--- a/src/AgentSquad.Runner/Components/StatusChart.razor
+++ b/src/AgentSquad.Runner/Components/StatusChart.razor
@@ -1,102 +1,54 @@
-@using AgentSquad.Runner.Models
+@using Microsoft.AspNetCore.Components
+@using Microsoft.JSInterop
 @implements IAsyncDisposable
 
-@inject IJSRuntime JSRuntime
-@inject ILogger<StatusChart> Logger
-
-<div class="status-chart-section">
-    <div class="status-chart-container">
-        <canvas @ref="chartCanvas" id="statusChart"></canvas>
+@if (StatusCounts.Shipped + StatusCounts.InProgress + StatusCounts.CarriedOver == 0)
+{
+    <div class="alert alert-info">No work items to display</div>
+}
+else
+{
+    <div class="chart-container">
+        <h3>Work Item Status Overview</h3>
+        <canvas id="statusChart"></canvas>
     </div>
-    
-    <div class="status-chart-legend">
-        <div class="status-chart-legend-item">
-            <div class="status-chart-legend-color shipped"></div>
-            <MudText Typo="Typo.caption">Shipped: @StatusCounts.Shipped</MudText>
-        </div>
-        <div class="status-chart-legend-item">
-            <div class="status-chart-legend-color in-progress"></div>
-            <MudText Typo="Typo.caption">In Progress: @StatusCounts.InProgress</MudText>
-        </div>
-        <div class="status-chart-legend-item">
-            <div class="status-chart-legend-color carried-over"></div>
-            <MudText Typo="Typo.caption">Carried Over: @StatusCounts.CarriedOver</MudText>
-        </div>
-    </div>
-</div>
+}
 
 @code {
     [Parameter]
     public (int Shipped, int InProgress, int CarriedOver) StatusCounts { get; set; }
 
-    private ElementReference chartCanvas;
-    private IJSObjectReference? chartModule;
-    private object? chartInstance;
+    [Inject]
+    public required IJSRuntime JSRuntime { get; set; }
 
-    protected override async Task OnParametersSetAsync()
-    {
-        try
-        {
-            if (chartCanvas.Id != null)
-            {
-                await InitializeChartAsync();
-            }
-        }
-        catch (Exception ex)
-        {
-            Logger.LogError(ex, "Error in OnParametersSetAsync");
-        }
-    }
+    private IJSObjectReference? module;
 
     protected override async Task OnAfterRenderAsync(bool firstRender)
     {
-        try
+        if (firstRender && StatusCounts.Shipped + StatusCounts.InProgress + StatusCounts.CarriedOver > 0)
         {
-            if (firstRender)
+            try
             {
-                await InitializeChartAsync();
+                module = await JSRuntime.InvokeAsync<IJSObjectReference>(
+                    "import", "./Components/StatusChart.razor.js");
+
+                await module.InvokeVoidAsync("initChart", 
+                    StatusCounts.Shipped, 
+                    StatusCounts.InProgress, 
+                    StatusCounts.CarriedOver);
             }
-        }
-        catch (Exception ex)
-        {
-            Logger.LogError(ex, "Error in OnAfterRenderAsync");
-        }
-    }
-
-    private async Task InitializeChartAsync()
-    {
-        try
-        {
-            chartModule ??= await JSRuntime.InvokeAsync<IJSObjectReference>(
-                "import", "./js/chart-helper.js");
-
-            chartInstance = await chartModule.InvokeAsync<object>(
-                "initializeChart",
-                chartCanvas,
-                new { 
-                    shipped = StatusCounts.Shipped,
-                    inProgress = StatusCounts.InProgress,
-                    carriedOver = StatusCounts.CarriedOver
-                });
-        }
-        catch (Exception ex)
-        {
-            Logger.LogError(ex, "Error initializing chart");
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error loading chart: {ex.Message}");
+            }
         }
     }
 
     async ValueTask IAsyncDisposable.DisposeAsync()
     {
-        try
+        if (module is not null)
         {
-            if (chartModule is not null)
-            {
-                await chartModule.DisposeAsync();
-            }
-        }
-        catch (Exception ex)
-        {
-            Logger.LogError(ex, "Error disposing chart module");
+            await module.DisposeAsync();
         }
     }
 }

--- a/src/AgentSquad.Runner/Components/WorkItemList.razor
+++ b/src/AgentSquad.Runner/Components/WorkItemList.razor
@@ -1,58 +1,39 @@
 @using AgentSquad.Runner.Models
+@using AgentSquad.Runner.Services
 
-<div class="work-item-sections">
-    @RenderWorkItemSection("Shipped", Shipped, "shipped")
-    @RenderWorkItemSection("In Progress", InProgress, "in-progress")
-    @RenderWorkItemSection("Carried Over", CarriedOver, "carried-over")
-</div>
+@if (Items == null || Items.Count == 0)
+{
+    <div class="alert alert-info">No work items in this category</div>
+}
+else
+{
+    <div class="work-items-container">
+        @foreach (var item in Items)
+        {
+            <div class="work-item-card">
+                <h4>@item.Title</h4>
+                <p><strong>Status:</strong> @GetStatusLabel(item.Status)</p>
+                @if (!string.IsNullOrEmpty(item.Assignee))
+                {
+                    <p><strong>Assignee:</strong> @item.Assignee</p>
+                }
+            </div>
+        }
+    </div>
+}
 
 @code {
     [Parameter]
-    public IReadOnlyList<WorkItem> Shipped { get; set; } = new List<WorkItem>();
+    public required IReadOnlyList<WorkItem> Items { get; set; }
 
-    [Parameter]
-    public IReadOnlyList<WorkItem> InProgress { get; set; } = new List<WorkItem>();
-
-    [Parameter]
-    public IReadOnlyList<WorkItem> CarriedOver { get; set; } = new List<WorkItem>();
-
-    private RenderFragment RenderWorkItemSection(string title, IReadOnlyList<WorkItem> items, string statusClass)
+    private string GetStatusLabel(WorkItemStatus status)
     {
-        return @<MudCard Class="work-item-section">
-            <MudCardHeader Class="@($"section-header {statusClass}")">
-                <CardHeaderContent>
-                    <div class="section-header-content">
-                        <MudText Typo="Typo.h6" Class="section-header-title">@title</MudText>
-                        <span class="section-header-count">@items.Count</span>
-                    </div>
-                </CardHeaderContent>
-            </MudCardHeader>
-            <MudCardContent>
-                @if (items.Count > 0)
-                {
-                    <ul class="work-item-list">
-                        @foreach (var item in items)
-                        {
-                            <li class="work-item">
-                                <h6 class="work-item-title">@item.Title</h6>
-                                @if (!string.IsNullOrEmpty(item.Assignee))
-                                {
-                                    <p class="assignee-text">@item.Assignee</p>
-                                }
-                            </li>
-                        }
-                    </ul>
-                }
-                else
-                {
-                    <div class="work-item-list-empty">
-                        <span class="work-item-list-empty-icon">✓</span>
-                        <MudText Typo="Typo.body2" Color="Color.TextSecondary">
-                            No items in this status
-                        </MudText>
-                    </div>
-                }
-            </MudCardContent>
-        </MudCard>;
+        return status switch
+        {
+            WorkItemStatus.Shipped => "Shipped",
+            WorkItemStatus.InProgress => "In Progress",
+            WorkItemStatus.CarriedOver => "Carried Over",
+            _ => "Unknown"
+        };
     }
 }

--- a/src/AgentSquad.Runner/Models/DashboardData.cs
+++ b/src/AgentSquad.Runner/Models/DashboardData.cs
@@ -2,60 +2,79 @@ using System.ComponentModel.DataAnnotations;
 
 namespace AgentSquad.Runner.Models;
 
-public enum WorkItemStatus
+public class DashboardData : IValidatableObject
 {
-    Shipped = 0,
-    InProgress = 1,
-    CarriedOver = 2
-}
-
-public class Project
-{
-    [Required(ErrorMessage = "Project name is required")]
-    [StringLength(256, ErrorMessage = "Project name cannot exceed 256 characters")]
-    public string Name { get; set; } = string.Empty;
-
-    [StringLength(1024, ErrorMessage = "Project description cannot exceed 1024 characters")]
-    public string? Description { get; set; }
-}
-
-public class Milestone
-{
-    [Required(ErrorMessage = "Milestone name is required")]
-    [StringLength(256, ErrorMessage = "Milestone name cannot exceed 256 characters")]
-    public string Name { get; set; } = string.Empty;
-
-    [Required(ErrorMessage = "Milestone date is required")]
-    public DateTime Date { get; set; }
-
-    [Required(ErrorMessage = "Milestone status is required")]
-    [RegularExpression(@"^(Completed|On Track|At Risk)$", 
-        ErrorMessage = "Milestone status must be one of: Completed, On Track, At Risk")]
-    public string Status { get; set; } = string.Empty;
-}
-
-public class WorkItem
-{
-    [Required(ErrorMessage = "Work item title is required")]
-    [StringLength(512, ErrorMessage = "Work item title cannot exceed 512 characters")]
-    public string Title { get; set; } = string.Empty;
-
-    [Required(ErrorMessage = "Work item status is required")]
-    [EnumDataType(typeof(WorkItemStatus), ErrorMessage = "Work item status must be Shipped, InProgress, or CarriedOver")]
-    public WorkItemStatus Status { get; set; }
-
-    [StringLength(256, ErrorMessage = "Assignee name cannot exceed 256 characters")]
-    public string? Assignee { get; set; }
-}
-
-public class DashboardData
-{
-    [Required(ErrorMessage = "Project is required")]
+    [Required(ErrorMessage = "Project is required.")]
     public Project? Project { get; set; }
 
-    [Required(ErrorMessage = "Milestones list is required")]
-    public List<Milestone> Milestones { get; set; } = [];
+    public List<Milestone> Milestones { get; set; } = new();
 
-    [Required(ErrorMessage = "Work items list is required")]
-    public List<WorkItem> WorkItems { get; set; } = [];
+    public List<WorkItem> WorkItems { get; set; } = new();
+
+    public IEnumerable<ValidationResult> Validate(ValidationContext context)
+    {
+        if (Project == null)
+        {
+            yield return new ValidationResult("Project cannot be null.", new[] { nameof(Project) });
+            yield break;
+        }
+
+        var projectResults = ((IValidatableObject)Project).Validate(context).ToList();
+        foreach (var result in projectResults)
+        {
+            yield return result;
+        }
+
+        if (Milestones == null)
+        {
+            yield return new ValidationResult("Milestones list cannot be null; use empty list instead.", new[] { nameof(Milestones) });
+        }
+        else
+        {
+            for (int i = 0; i < Milestones.Count; i++)
+            {
+                var milestone = Milestones[i];
+                if (milestone == null)
+                {
+                    yield return new ValidationResult($"Milestone at index {i} is null.", new[] { nameof(Milestones) });
+                    continue;
+                }
+
+                var milestoneResults = ((IValidatableObject)milestone).Validate(context).ToList();
+                foreach (var result in milestoneResults)
+                {
+                    var newMemberNames = result.MemberNames.Count() > 0
+                        ? result.MemberNames.Select(m => $"{nameof(Milestones)}[{i}].{m}").ToList()
+                        : new List<string> { $"{nameof(Milestones)}[{i}]" };
+                    yield return new ValidationResult(result.ErrorMessage, newMemberNames);
+                }
+            }
+        }
+
+        if (WorkItems == null)
+        {
+            yield return new ValidationResult("WorkItems list cannot be null; use empty list instead.", new[] { nameof(WorkItems) });
+        }
+        else
+        {
+            for (int i = 0; i < WorkItems.Count; i++)
+            {
+                var workItem = WorkItems[i];
+                if (workItem == null)
+                {
+                    yield return new ValidationResult($"WorkItem at index {i} is null.", new[] { nameof(WorkItems) });
+                    continue;
+                }
+
+                var workItemResults = ((IValidatableObject)workItem).Validate(context).ToList();
+                foreach (var result in workItemResults)
+                {
+                    var newMemberNames = result.MemberNames.Count() > 0
+                        ? result.MemberNames.Select(m => $"{nameof(WorkItems)}[{i}].{m}").ToList()
+                        : new List<string> { $"{nameof(WorkItems)}[{i}]" };
+                    yield return new ValidationResult(result.ErrorMessage, newMemberNames);
+                }
+            }
+        }
+    }
 }

--- a/src/AgentSquad.Runner/Models/Milestone.cs
+++ b/src/AgentSquad.Runner/Models/Milestone.cs
@@ -1,8 +1,40 @@
+using System.ComponentModel.DataAnnotations;
+
 namespace AgentSquad.Runner.Models;
 
-public class Milestone
+public class Milestone : IValidatableObject
 {
-    public string Name { get; set; }
+    private static readonly string[] ValidStatuses = { "Completed", "On Track", "At Risk" };
+
+    [Required(ErrorMessage = "Milestone name is required.")]
+    [StringLength(256, MinimumLength = 1, ErrorMessage = "Milestone name must be between 1 and 256 characters.")]
+    public string Name { get; set; } = string.Empty;
+
+    [Required(ErrorMessage = "Milestone date is required.")]
     public DateTime Date { get; set; }
-    public string Status { get; set; }
+
+    [Required(ErrorMessage = "Milestone status is required.")]
+    [RegularExpression(@"^(Completed|On Track|At Risk)$", 
+        ErrorMessage = "Milestone status must be 'Completed', 'On Track', or 'At Risk'.")]
+    public string Status { get; set; } = string.Empty;
+
+    public IEnumerable<ValidationResult> Validate(ValidationContext context)
+    {
+        if (string.IsNullOrWhiteSpace(Name))
+        {
+            yield return new ValidationResult("Milestone name cannot be empty or whitespace.", new[] { nameof(Name) });
+        }
+
+        if (Date == default)
+        {
+            yield return new ValidationResult("Milestone date cannot be the default DateTime value.", new[] { nameof(Date) });
+        }
+
+        if (!string.IsNullOrEmpty(Status) && !ValidStatuses.Contains(Status))
+        {
+            yield return new ValidationResult(
+                $"Milestone status '{Status}' is not valid. Valid values are: {string.Join(", ", ValidStatuses)}.",
+                new[] { nameof(Status) });
+        }
+    }
 }

--- a/src/AgentSquad.Runner/Models/Project.cs
+++ b/src/AgentSquad.Runner/Models/Project.cs
@@ -1,7 +1,26 @@
+using System.ComponentModel.DataAnnotations;
+
 namespace AgentSquad.Runner.Models;
 
-public class Project
+public class Project : IValidatableObject
 {
-    public string Name { get; set; }
-    public string Description { get; set; }
+    [Required(ErrorMessage = "Project name is required.")]
+    [StringLength(256, MinimumLength = 1, ErrorMessage = "Project name must be between 1 and 256 characters.")]
+    public string Name { get; set; } = string.Empty;
+
+    [StringLength(1024, ErrorMessage = "Project description cannot exceed 1024 characters.")]
+    public string? Description { get; set; }
+
+    public IEnumerable<ValidationResult> Validate(ValidationContext context)
+    {
+        if (string.IsNullOrWhiteSpace(Name))
+        {
+            yield return new ValidationResult("Project name cannot be empty or whitespace.", new[] { nameof(Name) });
+        }
+
+        if (Description != null && string.IsNullOrWhiteSpace(Description))
+        {
+            yield return new ValidationResult("Project description cannot be empty whitespace; use null instead.", new[] { nameof(Description) });
+        }
+    }
 }

--- a/src/AgentSquad.Runner/Models/WorkItem.cs
+++ b/src/AgentSquad.Runner/Models/WorkItem.cs
@@ -1,8 +1,36 @@
+using System.ComponentModel.DataAnnotations;
+
 namespace AgentSquad.Runner.Models;
 
-public class WorkItem
+public class WorkItem : IValidatableObject
 {
-    public string Title { get; set; }
+    [Required(ErrorMessage = "Work item title is required.")]
+    [StringLength(512, MinimumLength = 1, ErrorMessage = "Work item title must be between 1 and 512 characters.")]
+    public string Title { get; set; } = string.Empty;
+
+    [Required(ErrorMessage = "Work item status is required.")]
     public WorkItemStatus Status { get; set; }
-    public string Assignee { get; set; }
+
+    [StringLength(256, ErrorMessage = "Assignee name cannot exceed 256 characters.")]
+    public string? Assignee { get; set; }
+
+    public IEnumerable<ValidationResult> Validate(ValidationContext context)
+    {
+        if (string.IsNullOrWhiteSpace(Title))
+        {
+            yield return new ValidationResult("Work item title cannot be empty or whitespace.", new[] { nameof(Title) });
+        }
+
+        if (!Enum.IsDefined(typeof(WorkItemStatus), Status))
+        {
+            yield return new ValidationResult(
+                $"Work item status value '{Status}' is not a valid WorkItemStatus. Valid values are: Shipped, InProgress, CarriedOver.",
+                new[] { nameof(Status) });
+        }
+
+        if (Assignee != null && string.IsNullOrWhiteSpace(Assignee))
+        {
+            yield return new ValidationResult("Assignee cannot be empty whitespace; use null instead.", new[] { nameof(Assignee) });
+        }
+    }
 }

--- a/src/AgentSquad.Runner/Pages/Dashboard.razor
+++ b/src/AgentSquad.Runner/Pages/Dashboard.razor
@@ -1,102 +1,89 @@
 @page "/"
-@page "/dashboard"
-@using AgentSquad.Runner.Components
 @using AgentSquad.Runner.Models
 @using AgentSquad.Runner.Services
-@implements IAsyncDisposable
+@using AgentSquad.Runner.Components
 
-@inject DashboardDataService DataService
-@inject ILogger<Dashboard> Logger
+<PageTitle>Executive Dashboard</PageTitle>
 
-<MudContainer MaxWidth="MaxWidth.Lg" Class="dashboard-container">
-    @if (!DataService.HasData && !string.IsNullOrEmpty(DataService.GetLastError()))
+<div class="dashboard-container">
+    <h1>Executive Dashboard</h1>
+
+    @if (!string.IsNullOrEmpty(_errorMessage))
     {
-        <MudAlert Severity="Severity.Error" Class="dashboard-section">
-            <strong>Error Loading Dashboard</strong>
-            <p>@DataService.GetLastError()</p>
-            <p style="font-size: 12px; color: #666;">Check data.json format and try again.</p>
-        </MudAlert>
+        <div class="alert alert-danger alert-dismissible fade show" role="alert">
+            <strong>Error:</strong> @_errorMessage
+            <button type="button" class="btn-close" @onclick="@(() => _errorMessage = null)"></button>
+        </div>
     }
 
-    @if (DataService.HasData)
+    @if (_isLoading)
     {
-        <MudCard Class="dashboard-section">
-            <MudCardContent>
-                <MudText Typo="Typo.h4" Class="mb-3">@(DataService.GetProject()?.Name ?? "Project Dashboard")</MudText>
-                @if (!string.IsNullOrEmpty(DataService.GetProject()?.Description))
+        <div class="spinner-border" role="status">
+            <span class="visually-hidden">Loading...</span>
+        </div>
+    }
+    else if (DataService.HasData)
+    {
+        var project = DataService.GetProject();
+        var milestones = DataService.GetMilestones();
+        var statusCounts = DataService.GetStatusCounts();
+
+        @if (project != null)
+        {
+            <div class="project-header">
+                <h2>@project.Name</h2>
+                @if (!string.IsNullOrEmpty(project.Description))
                 {
-                    <MudText Typo="Typo.body1" Color="Color.TextSecondary">
-                        @DataService.GetProject().Description
-                    </MudText>
+                    <p class="description">@project.Description</p>
                 }
-            </MudCardContent>
-        </MudCard>
+            </div>
+        }
 
-        <MilestoneTimeline Milestones="DataService.GetMilestones()" />
+        <div class="status-overview">
+            <StatusChart StatusCounts="statusCounts"></StatusChart>
+        </div>
 
-        <MudCard Class="dashboard-section">
-            <MudCardHeader>
-                <CardHeaderContent>
-                    <MudText Typo="Typo.h5">Work Item Status</MudText>
-                </CardHeaderContent>
-            </MudCardHeader>
-            <MudCardContent>
-                <StatusChart StatusCounts="DataService.GetStatusCounts()" />
-            </MudCardContent>
-        </MudCard>
+        <div class="milestones-section">
+            <h3>Milestones</h3>
+            @if (milestones.Count == 0)
+            {
+                <p>No milestones defined</p>
+            }
+            else
+            {
+                <div class="milestones-list">
+                    @foreach (var milestone in milestones)
+                    {
+                        <div class="milestone-card">
+                            <h4>@milestone.Name</h4>
+                            <p><strong>Date:</strong> @milestone.Date.ToString("yyyy-MM-dd")</p>
+                            <p><strong>Status:</strong> @milestone.Status</p>
+                        </div>
+                    }
+                </div>
+            }
+        </div>
 
-        <WorkItemList 
-            Shipped="DataService.GetWorkItemsByStatus(WorkItemStatus.Shipped)"
-            InProgress="DataService.GetWorkItemsByStatus(WorkItemStatus.InProgress)"
-            CarriedOver="DataService.GetWorkItemsByStatus(WorkItemStatus.CarriedOver)" />
+        <div class="work-items-section">
+            <h3>Work Items by Status</h3>
+            <div class="row">
+                <div class="col-md-4">
+                    <h4>Shipped (@statusCounts.Shipped)</h4>
+                    <WorkItemList Items="DataService.GetWorkItemsByStatus(WorkItemStatus.Shipped)"></WorkItemList>
+                </div>
+                <div class="col-md-4">
+                    <h4>In Progress (@statusCounts.InProgress)</h4>
+                    <WorkItemList Items="DataService.GetWorkItemsByStatus(WorkItemStatus.InProgress)"></WorkItemList>
+                </div>
+                <div class="col-md-4">
+                    <h4>Carried Over (@statusCounts.CarriedOver)</h4>
+                    <WorkItemList Items="DataService.GetWorkItemsByStatus(WorkItemStatus.CarriedOver)"></WorkItemList>
+                </div>
+            </div>
+        </div>
     }
-    else if (DataService.HasData == false)
+    else
     {
-        <MudAlert Severity="Severity.Info" Class="dashboard-section">
-            <strong>Loading Dashboard...</strong>
-            <p>Please wait while the dashboard data is being loaded.</p>
-        </MudAlert>
+        <div class="alert alert-warning">No dashboard data available. Please check the data.json file.</div>
     }
-</MudContainer>
-
-@code {
-    protected override async Task OnInitializedAsync()
-    {
-        try
-        {
-            DataService.OnDataChanged += OnDataChanged;
-            await Task.Delay(100);
-            StateHasChanged();
-        }
-        catch (Exception ex)
-        {
-            Logger.LogError(ex, "Error initializing Dashboard component");
-        }
-    }
-
-    private void OnDataChanged()
-    {
-        try
-        {
-            InvokeAsync(() => StateHasChanged());
-        }
-        catch (Exception ex)
-        {
-            Logger.LogError(ex, "Error in OnDataChanged handler");
-        }
-    }
-
-    async ValueTask IAsyncDisposable.DisposeAsync()
-    {
-        try
-        {
-            DataService.OnDataChanged -= OnDataChanged;
-        }
-        catch (Exception ex)
-        {
-            Logger.LogError(ex, "Error disposing Dashboard component");
-        }
-        
-        await ValueTask.CompletedTask;
-    }
-}
+</div>

--- a/src/AgentSquad.Runner/Pages/Dashboard.razor.cs
+++ b/src/AgentSquad.Runner/Pages/Dashboard.razor.cs
@@ -1,35 +1,42 @@
-using AgentSquad.Runner.Models;
-using AgentSquad.Runner.Services;
 using Microsoft.AspNetCore.Components;
+using AgentSquad.Runner.Services;
 
-namespace AgentSquad.Runner.Pages
+namespace AgentSquad.Runner.Pages;
+
+public partial class Dashboard : ComponentBase, IAsyncDisposable
 {
-    public partial class Dashboard : ComponentBase, IAsyncDisposable
+    [Inject]
+    public required IDashboardDataService DataService { get; set; }
+
+    private string? _errorMessage;
+    private bool _isLoading = true;
+
+    protected override async Task OnInitializedAsync()
     {
-        [Inject]
-        public DashboardDataService DataService { get; set; }
-
-        protected override async Task OnInitializedAsync()
+        _isLoading = true;
+        
+        if (DataService.GetCurrentData() == null && DataService.GetLastError() != null)
         {
-            if (DataService != null)
-            {
-                DataService.OnDataChanged += OnDataChanged;
-            }
-            await base.OnInitializedAsync();
+            _errorMessage = DataService.GetLastError();
         }
 
-        private void OnDataChanged()
-        {
-            StateHasChanged();
-        }
+        DataService.OnDataChanged += OnDataChanged;
+        _isLoading = false;
+        await Task.CompletedTask;
+    }
 
-        async ValueTask IAsyncDisposable.DisposeAsync()
+    private void OnDataChanged()
+    {
+        _errorMessage = null;
+        StateHasChanged();
+    }
+
+    async ValueTask IAsyncDisposable.DisposeAsync()
+    {
+        if (DataService != null)
         {
-            if (DataService != null)
-            {
-                DataService.OnDataChanged -= OnDataChanged;
-            }
-            await Task.CompletedTask;
+            DataService.OnDataChanged -= OnDataChanged;
         }
+        await ValueTask.CompletedTask;
     }
 }

--- a/src/AgentSquad.Runner/Program.cs
+++ b/src/AgentSquad.Runner/Program.cs
@@ -1,16 +1,25 @@
-using AgentSquad.Runner.Models;
 using MudBlazor.Services;
+using AgentSquad.Runner.Models;
+using AgentSquad.Runner.Services;
 
 var builder = WebApplication.CreateBuilder(args);
 
 builder.Services.AddRazorPages();
 builder.Services.AddServerSideBlazor();
-
 builder.Services.AddMudBlazorSnackbar();
-builder.Services.AddMudBlazorDialog();
 
 builder.Services.Configure<DashboardOptions>(
     builder.Configuration.GetSection("Dashboard"));
+
+builder.Services.AddSingleton<IDashboardDataService, DashboardDataService>();
+
+builder.Services.AddMudBlazorDialog();
+
+builder.Services.AddLogging(config =>
+{
+    config.AddConsole();
+    config.AddDebug();
+});
 
 var app = builder.Build();
 
@@ -21,9 +30,7 @@ if (!app.Environment.IsDevelopment())
 }
 
 app.UseHttpsRedirection();
-
 app.UseStaticFiles();
-
 app.UseRouting();
 
 app.MapBlazorHub();

--- a/src/AgentSquad.Runner/Services/DashboardDataService.cs
+++ b/src/AgentSquad.Runner/Services/DashboardDataService.cs
@@ -1,61 +1,127 @@
+using System.ComponentModel.DataAnnotations;
 using System.Security.Cryptography;
 using System.Text;
 using System.Text.Json;
-using AgentSquad.Runner.Models;
+using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
+using AgentSquad.Runner.Models;
 
 namespace AgentSquad.Runner.Services;
 
-public class DashboardDataService : IDisposable
+public class DashboardDataService : IDashboardDataService, IDisposable
 {
     private readonly ILogger<DashboardDataService> _logger;
     private readonly IOptions<DashboardOptions> _options;
-    private readonly FileSystemWatcher? _watcher;
     private DashboardData? _cachedData;
-    private string? _lastLoadedHash;
     private string? _lastError;
-    private Timer? _debounceTimer;
-    private bool _isDisposed;
+    private string? _lastLoadedHash;
+    private FileSystemWatcher? _fileWatcher;
+    private System.Threading.Timer? _debounceTimer;
+    private bool _disposed;
 
     public event Action? OnDataChanged;
 
-    public bool HasData => _cachedData != null && string.IsNullOrEmpty(_lastError);
-
-    public DashboardDataService(ILogger<DashboardDataService> logger, IOptions<DashboardOptions> options)
+    public DashboardDataService(
+        ILogger<DashboardDataService> logger,
+        IOptions<DashboardOptions> options)
     {
-        _logger = logger;
-        _options = options;
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        _options = options ?? throw new ArgumentNullException(nameof(options));
 
+        LoadData();
+        InitializeFileWatcher();
+    }
+
+    public DashboardData? GetCurrentData() => _cachedData;
+
+    public Project? GetProject() => _cachedData?.Project;
+
+    public IReadOnlyList<Milestone> GetMilestones()
+    {
+        if (_cachedData?.Milestones == null || _cachedData.Milestones.Count == 0)
+        {
+            return new List<Milestone>().AsReadOnly();
+        }
+
+        return _cachedData.Milestones
+            .OrderBy(m => m.Date)
+            .ToList()
+            .AsReadOnly();
+    }
+
+    public IReadOnlyList<WorkItem> GetWorkItems()
+    {
+        if (_cachedData?.WorkItems == null || _cachedData.WorkItems.Count == 0)
+        {
+            return new List<WorkItem>().AsReadOnly();
+        }
+
+        return _cachedData.WorkItems.AsReadOnly();
+    }
+
+    public IReadOnlyList<WorkItem> GetWorkItemsByStatus(WorkItemStatus status)
+    {
+        if (_cachedData?.WorkItems == null || _cachedData.WorkItems.Count == 0)
+        {
+            return new List<WorkItem>().AsReadOnly();
+        }
+
+        return _cachedData.WorkItems
+            .Where(w => w.Status == status)
+            .ToList()
+            .AsReadOnly();
+    }
+
+    public (int Shipped, int InProgress, int CarriedOver) GetStatusCounts()
+    {
+        if (_cachedData?.WorkItems == null || _cachedData.WorkItems.Count == 0)
+        {
+            return (0, 0, 0);
+        }
+
+        int shipped = _cachedData.WorkItems.Count(w => w.Status == WorkItemStatus.Shipped);
+        int inProgress = _cachedData.WorkItems.Count(w => w.Status == WorkItemStatus.InProgress);
+        int carriedOver = _cachedData.WorkItems.Count(w => w.Status == WorkItemStatus.CarriedOver);
+
+        return (shipped, inProgress, carriedOver);
+    }
+
+    public string? GetLastError() => _lastError;
+
+    public bool HasData => _cachedData != null;
+
+    private void LoadData()
+    {
         try
         {
-            LoadDataFromFile();
-            InitializeFileWatcher();
-            _logger.LogInformation("DashboardDataService initialized successfully");
+            string dataJsonPath = ResolveDataJsonPath();
+
+            if (!File.Exists(dataJsonPath))
+            {
+                SetError($"data.json not found at {dataJsonPath}. Please create the file or update the DataJsonPath in appsettings.json.");
+                _logger.LogError($"Data file not found at path: {dataJsonPath}");
+                return;
+            }
+
+            string json = ReadFileWithRetry(dataJsonPath);
+            LoadFromJson(json);
+            _logger.LogInformation($"DashboardDataService initialized successfully. Data loaded from {dataJsonPath}");
         }
         catch (Exception ex)
         {
-            _logger.LogError(ex, "Error initializing DashboardDataService");
-            _lastError = $"Failed to initialize dashboard: {ex.Message}";
+            SetError($"Unexpected error loading dashboard data: {ex.Message}");
+            _logger.LogError(ex, "Unexpected error in DashboardDataService initialization");
         }
     }
 
-    private void LoadDataFromFile()
+    private void LoadFromJson(string json)
     {
         try
         {
-            var filePath = Path.Combine(AppContext.BaseDirectory, _options.Value.DataJsonPath);
-
-            if (!File.Exists(filePath))
+            if (string.IsNullOrWhiteSpace(json))
             {
-                throw new FileNotFoundException($"data.json not found at {filePath}");
-            }
-
-            var json = File.ReadAllText(filePath, Encoding.UTF8);
-            var fileHash = ComputeHash(json);
-
-            if (fileHash == _lastLoadedHash)
-            {
-                _logger.LogDebug("File hash unchanged; skipping reload");
+                SetError("data.json file is empty. Please provide valid JSON content.");
+                _logger.LogError("data.json file is empty");
                 return;
             }
 
@@ -66,144 +132,173 @@ public class DashboardDataService : IDisposable
                 MaxDepth = 64
             };
 
-            _cachedData = JsonSerializer.Deserialize<DashboardData>(json, options)
-                ?? throw new JsonException("Failed to deserialize data.json");
+            DashboardData? data = JsonSerializer.Deserialize<DashboardData>(json, options);
 
-            ValidateData(_cachedData);
-            _lastLoadedHash = fileHash;
+            ValidateData(data);
+
+            _cachedData = data;
             _lastError = null;
-
-            _logger.LogInformation("data.json loaded and parsed successfully");
+            _lastLoadedHash = ComputeHash(json);
+            _logger.LogInformation("Dashboard data loaded and validated successfully");
         }
         catch (JsonException ex)
         {
-            _lastError = $"JSON parsing error: {ex.Message}";
-            _logger.LogError(ex, "Failed to parse data.json");
+            SetError($"JSON syntax error in data.json: {ex.Message}");
+            _logger.LogError(ex, "JSON parsing error in data.json");
         }
-        catch (IOException ex)
+        catch (ValidationException ex)
         {
-            _lastError = $"File read error: {ex.Message}";
-            _logger.LogError(ex, "IOException reading data.json");
+            SetError($"Data validation error: {ex.Message}");
+            _logger.LogError(ex, "Data validation failed");
         }
-        catch (Exception ex)
+    }
+
+    private void ValidateData(DashboardData? data)
+    {
+        if (data == null)
         {
-            _lastError = $"Unexpected error: {ex.Message}";
-            _logger.LogError(ex, "Unexpected error loading data.json");
+            throw new ValidationException("Parsed data is null. Please provide valid JSON matching the expected schema.");
         }
+
+        var context = new ValidationContext(data);
+        var results = new List<ValidationResult>();
+        bool isValid = Validator.TryValidateObject(data, context, results, validateAllProperties: true);
+
+        if (!isValid)
+        {
+            var errorMessages = results
+                .Select(r => r.ErrorMessage ?? "Validation failed with no error message")
+                .ToList();
+
+            string combinedError = string.Join(" | ", errorMessages);
+            throw new ValidationException($"Data validation failed: {combinedError}");
+        }
+    }
+
+    private string ReadFileWithRetry(string path, int maxRetries = 3)
+    {
+        for (int i = 0; i < maxRetries; i++)
+        {
+            try
+            {
+                return File.ReadAllText(path, Encoding.UTF8);
+            }
+            catch (IOException ex) when (i < maxRetries - 1)
+            {
+                _logger.LogWarning($"File locked; retrying in 100ms ({i + 1}/{maxRetries})");
+                Thread.Sleep(100);
+            }
+            catch (IOException ex) when (i == maxRetries - 1)
+            {
+                throw new IOException($"Could not read {path} after {maxRetries} retries: {ex.Message}", ex);
+            }
+        }
+
+        throw new IOException($"Could not read {path} after {maxRetries} retries");
+    }
+
+    private string ResolveDataJsonPath()
+    {
+        string? configuredPath = _options.Value?.DataJsonPath;
+
+        if (string.IsNullOrWhiteSpace(configuredPath))
+        {
+            configuredPath = "data.json";
+        }
+
+        if (Path.IsPathRooted(configuredPath))
+        {
+            return configuredPath;
+        }
+
+        return Path.Combine(AppContext.BaseDirectory, configuredPath);
+    }
+
+    private string ComputeHash(string content)
+    {
+        using (var sha256 = SHA256.Create())
+        {
+            byte[] hashBytes = sha256.ComputeHash(Encoding.UTF8.GetBytes(content));
+            return Convert.ToBase64String(hashBytes);
+        }
+    }
+
+    private void SetError(string message)
+    {
+        _lastError = message;
+        _cachedData = null;
     }
 
     private void InitializeFileWatcher()
     {
         try
         {
-            var directory = AppContext.BaseDirectory;
-            var fileName = _options.Value.DataJsonPath;
+            string dataJsonPath = ResolveDataJsonPath();
+            string? directory = Path.GetDirectoryName(dataJsonPath);
+            string? fileName = Path.GetFileName(dataJsonPath);
 
-            _watcher = new FileSystemWatcher(directory)
+            if (string.IsNullOrEmpty(directory) || string.IsNullOrEmpty(fileName))
             {
-                Filter = fileName,
+                _logger.LogWarning("Could not initialize file watcher: invalid path");
+                return;
+            }
+
+            _fileWatcher = new FileSystemWatcher(directory, fileName)
+            {
                 NotifyFilter = NotifyFilters.LastWrite | NotifyFilters.Size,
                 EnableRaisingEvents = true
             };
 
-            _watcher.Changed += OnFileChanged;
-            _logger.LogInformation("FileSystemWatcher initialized for data.json");
+            _fileWatcher.Changed += OnFileChanged;
+            _logger.LogInformation($"File watcher initialized for {dataJsonPath}");
         }
         catch (Exception ex)
         {
-            _logger.LogError(ex, "Error initializing FileSystemWatcher");
+            _logger.LogWarning(ex, "Failed to initialize file watcher");
         }
     }
 
     private void OnFileChanged(object sender, FileSystemEventArgs e)
     {
-        try
-        {
-            _debounceTimer?.Dispose();
-            _debounceTimer = new Timer(
-                _ => ReloadData(),
-                null,
-                _options.Value.FileWatchDebounceMs,
-                Timeout.Infinite);
-        }
-        catch (Exception ex)
-        {
-            _logger.LogError(ex, "Error in OnFileChanged");
-        }
+        _debounceTimer?.Dispose();
+        _debounceTimer = new System.Threading.Timer(
+            _ => ReloadData(),
+            null,
+            _options.Value?.FileWatchDebounceMs ?? 500,
+            System.Threading.Timeout.Infinite);
     }
 
     private void ReloadData()
     {
         try
         {
-            _debounceTimer?.Dispose();
-            LoadDataFromFile();
-            OnDataChanged?.Invoke();
+            string dataJsonPath = ResolveDataJsonPath();
+            string json = ReadFileWithRetry(dataJsonPath);
+            string newHash = ComputeHash(json);
+
+            if (newHash != _lastLoadedHash)
+            {
+                LoadFromJson(json);
+                OnDataChanged?.Invoke();
+                _logger.LogInformation("Dashboard data reloaded from file system watcher");
+            }
         }
         catch (Exception ex)
         {
-            _logger.LogError(ex, "Error reloading data");
+            _logger.LogError(ex, "Error reloading dashboard data from file watcher");
         }
     }
 
-    private void ValidateData(DashboardData data)
-    {
-        if (data?.Project == null)
-            throw new InvalidOperationException("Project is required");
-
-        if (string.IsNullOrEmpty(data.Project.Name))
-            throw new InvalidOperationException("Project name is required");
-
-        if (data.Milestones?.Any(m => string.IsNullOrEmpty(m.Name)) == true)
-            throw new InvalidOperationException("Milestone name is required");
-
-        if (data.WorkItems?.Any(w => !Enum.IsDefined(w.Status)) == true)
-            throw new InvalidOperationException("Invalid work item status");
-    }
-
-    private string ComputeHash(string input)
-    {
-        using var sha = SHA256.Create();
-        var hash = sha.ComputeHash(Encoding.UTF8.GetBytes(input));
-        return Convert.ToBase64String(hash);
-    }
-
-    public DashboardData? GetCurrentData() => _cachedData;
-
-    public Project? GetProject() => _cachedData?.Project;
-
-    public IReadOnlyList<Milestone> GetMilestones() =>
-        _cachedData?.Milestones?.OrderBy(m => m.Date).ToList().AsReadOnly()
-        ?? new List<Milestone>().AsReadOnly();
-
-    public IReadOnlyList<WorkItem> GetWorkItems() =>
-        _cachedData?.WorkItems?.AsReadOnly() ?? new List<WorkItem>().AsReadOnly();
-
-    public IReadOnlyList<WorkItem> GetWorkItemsByStatus(WorkItemStatus status) =>
-        _cachedData?.WorkItems?
-            .Where(w => w.Status == status)
-            .ToList()
-            .AsReadOnly() ?? new List<WorkItem>().AsReadOnly();
-
-    public (int Shipped, int InProgress, int CarriedOver) GetStatusCounts()
-    {
-        var items = _cachedData?.WorkItems ?? new List<WorkItem>();
-        return (
-            items.Count(w => w.Status == WorkItemStatus.Shipped),
-            items.Count(w => w.Status == WorkItemStatus.InProgress),
-            items.Count(w => w.Status == WorkItemStatus.CarriedOver)
-        );
-    }
-
-    public string? GetLastError() => _lastError;
-
     public void Dispose()
     {
-        if (_isDisposed)
+        if (_disposed)
+        {
             return;
+        }
 
-        _watcher?.Dispose();
+        _fileWatcher?.Dispose();
         _debounceTimer?.Dispose();
-        _isDisposed = true;
+        _disposed = true;
+        GC.SuppressFinalize(this);
     }
 }

--- a/src/AgentSquad.Runner/Services/IDashboardDataService.cs
+++ b/src/AgentSquad.Runner/Services/IDashboardDataService.cs
@@ -4,7 +4,7 @@ namespace AgentSquad.Runner.Services;
 
 public interface IDashboardDataService
 {
-    event Action OnDataChanged;
+    event Action? OnDataChanged;
 
     DashboardData? GetCurrentData();
     Project? GetProject();

